### PR TITLE
Tril Improvements

### PR DESCRIPTION
**Pull request by Leonardo2718**
This set of commits brings some major improvements to Tril, including:

1. Restructuring of the project into logical components
2. Addition of a (small) GTest-based test suite
3. Improved documentation

Even more improvements are coming in a PR I will be opening soon after this one is merged.

### CODE REVIEW

File: fvtest/tril/README.md [hunk 1]
```diff
@@ -1,34 +1,75 @@
...
+Intermediate Language (IL). Its goal is to simplify testing Testarossa by
 providing very fine control over the IL that is fed into the compiler.
 
-## trilc
+## Overview
 
-trilc is an experimental compiler that reads a Tril file, compiles the contained
-code, and executes it.
+The Tril implementation is a small library with a parser and IL generator,
+located in the `tirl/` directory. It provides the core tools for dealing with
```
Comment (mgaudet) [inline fvtest/tril/README.md]: typo; `tril`

**Comment by Leonardo2718** [comment]
@mgaudet That's correct.

The big problem is that `trilc` used a function pointer to call the JITed code. This meant that it could only compile/call methods with a specific signature, which is not very flexible.

The new examples work essentially the same as `trilc` did however. They just don't try to pretend they can do anything other than what they were made for :smile: .

**Comment by mgaudet** [comment]
:+1:  LGTM

File: fvtest/tril/README.md (hunk 1 above)
Comment (Leonardo2718) [inline fvtest/tril/README.md]: fixed.

File: fvtest/tril/README.md [hunk 2]
```diff
@@ -1,34 +1,75 @@
...
-trilc is an experimental compiler that reads a Tril file, compiles the contained
-code, and executes it.
+The Tril implementation is a small library with a parser and IL generator,
+located in the `tril/` directory. It provides the core tools for dealing with
+Tril code. Compilation and executing of Tril code is implemented separately.
+This allows custom tools to be easily developed on top of Tril.
 
-Although it is still in an experimental stage, trilc is already capable of
-representing fairly complex programs (see `mandelbrot.tril` example).
+The `examples/` directory contains examples demonstrating the how to implement
```
Comment (0xdaryl) [inline fvtest/tril/README.md]: delete 'the'

File: fvtest/tril/README.md [hunk 3]
```diff
@@ -1,34 +1,75 @@
...
 
-## Building and running
+The `test/` directory contains some GTest-based test cases for Tril.
 
-1) Make sure you have cmake, flex, and yacc installed on your machine
-2) Clone the Eclipse OMR repo: `git clone https://github.com/eclipse/omr.git`
-3) execute the following commands in a terminal:
+## Building Tril
+
+1. Make sure you have cmake, flex, and yacc installed on your machine
```
Comment (0xdaryl) [inline fvtest/tril/README.md]: Any particular version required?

File: fvtest/tril/test/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,50 @@
+###############################################################################
+#
+# (c) Copyright IBM Corp. 2017
```
Comment (0xdaryl) [inline fvtest/tril/test/CMakeLists.txt]: 2017, 2017

**Comment by Leonardo2718** [comment]
@0xdaryl fixed.

**Pull request merged**